exporter/tanzuobservabilityexporter/metrics_test.go

```diff
@@ -197,6 +196,223 @@ func TestGaugeConsumerMissingValue(t *testing.T) {
+			Value: 52.375,
+			Tags:  map[string]string{"env": "prod"},
+		},
+	}
+	assert.ElementsMatch(t, expected, sender.deltaMetrics)
+	assert.Empty(t, errs)
+}
+
+func TestSumConsumerErrorOnSend(t *testing.T) {
+	deltaMetric := newMetric(
+		"test.delta.metric", pdata.MetricDataTypeSum)
+	sum := deltaMetric.Sum()
+	sum.SetAggregationTemporality(pdata.MetricAggregationTemporalityDelta)
+	dataPoints := sum.DataPoints()
+	dataPoints.EnsureCapacity(2)
+	addDataPoint(
+		35,
+		1635205001,
+		map[string]interface{}{
+			"env": "dev",
+		},
+		dataPoints,
+	)
+	addDataPoint(
+		52.375,
+		1635205002,
+		map[string]interface{}{
+			"env": "prod",
+		},
+		dataPoints,
+	)
+
+	sender := &mockSumSender{errorOnSend: true}
+	consumer := newSumConsumer(sender, nil)
+	assert.Equal(t, pdata.MetricDataTypeSum, consumer.Type())
+	var errs []error
+
+	// delta sums get treated as delta counters
+	consumer.Consume(deltaMetric, &errs)
+	consumer.PushInternalMetrics(&errs)
+
+	assert.Len(t, errs, 2)
+}
+
+func TestSumConsumerCumulative(t *testing.T) {
+	cumulativeMetric := newMetric(
+		"test.cumulative.metric", pdata.MetricDataTypeSum)
+	sum := cumulativeMetric.Sum()
+	sum.SetAggregationTemporality(pdata.MetricAggregationTemporalityCumulative)
+	dataPoints := sum.DataPoints()
+	dataPoints.EnsureCapacity(1)
+	addDataPoint(
+		62.25,
+		1634205001,
+		map[string]interface{}{
+			"env": "dev",
+		},
+		dataPoints,
+	)
+	sender := &mockSumSender{}
+	consumer := newSumConsumer(sender, nil)
+	assert.Equal(t, pdata.MetricDataTypeSum, consumer.Type())
+	var errs []error
+
+	// cumulative sums get treated as regular wavefront metrics
+	consumer.Consume(cumulativeMetric, &errs)
+	consumer.PushInternalMetrics(&errs)
+
+	expected := []tobsMetric{
+		{
+			Name:  "test.cumulative.metric",
+			Value: 62.25,
+			Ts:    1634205001,
+			Tags:  map[string]string{"env": "dev"},
+		},
+	}
+	assert.ElementsMatch(t, expected, sender.metrics)
+	assert.Empty(t, errs)
+}
+
+func TestSumConsumerUnspecified(t *testing.T) {
+	cumulativeMetric := newMetric(
+		"test.unspecified.metric", pdata.MetricDataTypeSum)
+	sum := cumulativeMetric.Sum()
+	sum.SetAggregationTemporality(pdata.MetricAggregationTemporalityUnspecified)
+	dataPoints := sum.DataPoints()
+	dataPoints.EnsureCapacity(1)
+	addDataPoint(
+		72.25,
+		1634206001,
+		map[string]interface{}{
+			"env": "qa",
+		},
+		dataPoints,
+	)
+	sender := &mockSumSender{}
+	consumer := newSumConsumer(sender, nil)
+	assert.Equal(t, pdata.MetricDataTypeSum, consumer.Type())
+	var errs []error
+
+	// unspecified sums get treated as regular wavefront metrics
+	consumer.Consume(cumulativeMetric, &errs)
+	consumer.PushInternalMetrics(&errs)
+
+	expected := []tobsMetric{
+		{
+			Name:  "test.unspecified.metric",
+			Value: 72.25,
+			Ts:    1634206001,
+			Tags:  map[string]string{"env": "qa"},
+		},
+	}
+	assert.ElementsMatch(t, expected, sender.metrics)
+	assert.Empty(t, errs)
+}
+
+func TestSumConsumerMissingValueNoLogging(t *testing.T) {
```

Comment:
I think we have pretty good coverage of the desired behavior for Missing Values in  `TestSumConsumerMissingValue`. Consider deleting this test altogether.

Reply:
Done.